src/extjson/de.rs

```diff
@@ -0,0 +1,230 @@
+//! Deserializing [MongoDB Extended JSON v2](https://docs.mongodb.com/manual/reference/mongodb-extended-json/)
```

Comment:
This module is largely empty for the moment, but in the future I imagine all the various `from_value` / `from_reader` / etc. functions for extjson could live here. Likewise, we could introduce a `ser` module that has the counterparts to those.

e.g.
```rust
pub(crate) fn from_value<T>(ext_json: serde_json::Value) -> Result<T>
where
    T: DeserializeOwned,
{
    let bson: crate::Bson = ext_json.try_into()?;
    let result = crate::from_bson(bson)?;
    Ok(result)
}
```